packages/airnode-node/src/handlers/sign-oev-data.ts

```diff
@@ -0,0 +1,85 @@
+import { ethers } from 'ethers';
+import { go } from '@api3/promise-utils';
+import { SignOevDataResponse } from '../types';
+import { getAirnodeWalletFromPrivateKey } from '../evm/wallet';
+import { ProcessSignOevDataRequestBody } from '../workers/local-gateways';
+import { getExpectedTemplateIdV1 } from '../evm/templates';
+
+export function deriveBeaconId(airnodeAddress: string, templateId: string) {
+  return ethers.utils.solidityKeccak256(['address', 'bytes32'], [airnodeAddress, templateId]);
+}
+
+export function deriveBeaconSetId(beaconIds: string[]) {
+  return ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['bytes32[]'], [beaconIds]));
+}
+
+export function calculateMedian(arr: ethers.BigNumber[]) {
+  const mid = Math.floor(arr.length / 2);
+  const nums = [...arr].sort((a, b) => {
+    if (a.lt(b)) return -1;
+    else if (a.gt(b)) return 1;
+    else return 0;
+  });
+  return arr.length % 2 !== 0 ? nums[mid] : nums[mid - 1].add(nums[mid]).div(2);
+}
+
+export async function signOevData(
+  requestBody: ProcessSignOevDataRequestBody,
+  validUpdateValues: ethers.BigNumber[]
+): Promise<[Error, null] | [null, SignOevDataResponse]> {
+  const { chainId, dapiServerAddress, oevProxyAddress, updateId, bidderAddress, bidAmount, signedData } = requestBody;
+  const airnodeWallet = getAirnodeWalletFromPrivateKey();
+  const airnodeAddress = airnodeWallet.address;
+  const timestamp = Math.floor(Date.now() / 1000).toString();
+
+  const beaconsWithTemplateId = signedData.map((beacon) => {
+    const templateId = getExpectedTemplateIdV1({
+      airnodeAddress: beacon.airnodeAddress,
+      endpointId: beacon.endpointId,
+      encodedParameters: beacon.encodedParameters,
+    });
+
+    return { ...beacon, templateId };
+  });
+  const beaconIds = beaconsWithTemplateId.map((beacon) => deriveBeaconId(beacon.airnodeAddress, beacon.templateId));
+  const dataFeedId = beaconIds.length === 1 ? beaconIds[0] : deriveBeaconSetId(beaconIds);
+  const updateValue = calculateMedian(validUpdateValues);
```

Comment:
This deserves a comment. We are computing this in Airnode otherwise it would be possible to spoof the signature.